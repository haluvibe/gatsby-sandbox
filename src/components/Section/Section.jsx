import React from 'react';
import Layout from '../Layout/Layout';
import uuidv4 from 'uuid/v4';

export default ({children, settings}) =>
  <section {...settings}>
    { children && children.map(component => <Layout key={uuidv4()} {...component} />) }
  </section>
