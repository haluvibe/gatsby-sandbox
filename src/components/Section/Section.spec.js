import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Section from './Section';

const mockSettings = {
  id: 'id',
  className: 'a-class another-class',
  style: {
    backgroundColor: 'red',
    fontSize: 16,
  }
}

test('Section component should render as expected', () => {
  const component = shallow(<Section settings={mockSettings} children={null} />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
