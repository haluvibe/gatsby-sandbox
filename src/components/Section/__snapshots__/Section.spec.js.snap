// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component should render as expected 1`] = `
<section
  className="a-class another-class"
  id="id"
  style={
    Object {
      "backgroundColor": "red",
      "fontSize": 16,
    }
  }
/>
`;
